aea/agent.py

```diff
@@ -206,8 +206,7 @@ def stop(self) -> None:
         self.teardown()
 
         logger.debug("[{}]: Stopping message processing...".format(self.name))
-        if self.multiplexer.connection_status.is_connected:
-            self.multiplexer.disconnect()
+        self.multiplexer.disconnect()
```

Comment:
we do the check inside `multiplexer.disconnect()`

Reply:
then it should be made symmetric for the connect